components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -264,7 +278,11 @@ func resendMessage(httpClient *http.Client, channel *evapisv1alpha1.Channel, hea
 				log.Printf("ERROR: resendMessage(): could not resend HTTP request: %v", err)
 				return err
 			}
-			defer func() { _ = res.Body.Close() }()
+			defer func() {
+				_ = res.Body.Close()
+				tran, _ := httpClient.Transport.(*http.Transport)
```

Comment:
I think `tran.CloseIdleConnections()` needs to be in a separate `defer` right after `httpClient.Do(req)`.
If `httpClient.Do(req)` returns an error then we wouldn't be closing idle connections.

Reply:
`defer()` should be called after `resendMessage()` exits, right?

Reply:
fixed